tests/wfqueue.test.ts

```diff
@@ -664,6 +664,107 @@ describe('queued-wf-tests-simple', () => {
   });
 });
 
+// dummy declaration to match the workflow in tests/wfqueueworker.ts
+export class InterProcessWorkflowTask {
+  @DBOS.workflow()
+  static async task() {
+    return Promise.resolve();
+  }
+}
+
+// This queue cannot dequeue
+const IPWQueue = new WorkflowQueue('IPWQueue', {
+  rateLimit: { limitPerPeriod: 0, periodSec: 30 },
+});
+class InterProcessWorkflow {
+  static globalConcurrencyLimit = 3;
+  static nWorkers = InterProcessWorkflow.globalConcurrencyLimit * 2;
+  static nTasks = InterProcessWorkflow.globalConcurrencyLimit * 5;
+
+  @DBOS.workflow()
+  static async testGlobalConcurrency() {
+    // Enqueue a bunch of tasks
+    const handles = [];
+    for (let i = 0; i < InterProcessWorkflow.nTasks; ++i) {
+      handles.push(await DBOS.startWorkflow(InterProcessWorkflowTask, { queueName: IPWQueue.name }).task());
+    }
+    // Start the processes and wait for the to notify us
+    const workerPromises = await InterProcessWorkflow.startWorkerProcesses();
+    try {
+      let num_dequeued = 0;
+      while (num_dequeued < InterProcessWorkflow.globalConcurrencyLimit) {
+        const msg = await DBOS.recv<string>('worker_dequeue', 1);
+        if (msg) {
+          num_dequeued++;
+          DBOS.logger.debug(`Dequeued ${num_dequeued} tasks`);
+        }
+      }
+      // Check the concurrency limit is respected
+      let dequeuedWorkflows = 0;
+      let enqueuedWorkflows = 0;
+      for (const h of handles) {
+        const status = await h.getStatus();
+        expect(status).not.toBeNull();
+        if (status?.status === StatusString.ENQUEUED) {
+          enqueuedWorkflows++;
+        } else if (status?.status === StatusString.PENDING) {
+          dequeuedWorkflows++;
+        }
+      }
+      expect(dequeuedWorkflows).toBe(InterProcessWorkflow.globalConcurrencyLimit);
+      expect(enqueuedWorkflows).toBe(InterProcessWorkflow.nTasks - InterProcessWorkflow.globalConcurrencyLimit);
+
+      // Notify the workers they can resume
+      await DBOS.setEvent('worker_resume', true);
+    } catch (e) {
+      DBOS.logger.error(`Error: ${(e as Error).message}`);
+      throw e;
+    } finally {
+      await Promise.all(workerPromises);
+    }
+  }
+
+  @DBOS.step()
+  static startWorkerProcesses(): Promise<Promise<void>[]> {
+    const workerPromises: Promise<void>[] = [];
+    for (let i = 0; i < InterProcessWorkflow.nWorkers; i++) {
+      const workerId = `test-worker-${i}`;
+      const workerPromise = new Promise<void>((resolve, reject) => {
+        const child = spawn(
```

Comment:
`spawn` let's you stream the children stdio in the calling process, which is very useful for debugging.